### PR TITLE
Adding getAllMatches Function

### PR DESCRIPTION
**Pull request by honjes**
Extracting Matches constructor into a function.
Adding getAllMatches Function to Matches
Changing URI-Schema from the getSingleMatch Function to accept only the id of an Match or the complete path.
Adding 2 new Test's

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -119,6 +125,46 @@ http://localhost:3000/results
 ```
 
 ##### Matches
+
+###### All Matches
+
+```js
+app.get('/allMatches', (req, res) => {
```
Comment (dajk) [inline README.md]: I'd prefer using `/all-matches` and encourage best practices, or rather more common solution.

File: examples/with-typescript/index.ts [hunk 2]
```diff
@@ -11,6 +11,10 @@ app.get('/results', (req, res) => {
   getResults((results: any) => res.json(results))
 })
 
+app.get('/allMatches', (req, res) => {
```
Comment (dajk) [inline examples/with-typescript/index.ts]: also here `/all-matches`

File: examples/with-babel/index.js [hunk 3]
```diff
@@ -11,6 +11,10 @@ app.get('/results', (req, res) => {
   hltv.getResults(results => res.json(results))
 })
 
+app.get('/allMatches', (req, res) => {
```
Comment (dajk) [inline examples/with-babel/index.js]: also here `/all-matches`

**Comment by dajk** [comment]
Thanks so much @honjes , I'd be more than happy to support you to make your first contribution.
I had the first round of checks, will check it again more in detail after the first iteration.

File: tests/index.test.ts [hunk 4]
```diff
@@ -35,7 +35,121 @@ describe('hltv-api', () => {
       })
     })
 
-    it('should have match stats when we call `getMatches`', done => {
+    it('should have match stats when we call `getMatches` with long Id', done => {
```
[the remark this answers is not in the document]
Comment (honjes) [reply tests/index.test.ts]: Yes its duplicatet but with the longer id schema. So its tests both behaviors.

**Comment by honjes** [comment]
It seems linke eslint dose not like it when you do not define a type.

**Comment by honjes** [comment]
I think that this commit is good. Do you have somthing to add ?

File: src/matches.ts [hunk 5]
```diff
@@ -18,7 +22,19 @@ export default class Matches {
...
+    if (matchId.split('/').length >= 3) {
+      this.matchId =
+        parseInt(matchId.split('/')[1], 10) > 100
+          ? matchId.split('/')[1]
+          : matchId.split('/')[2]
+    } else {
+      this.matchId = matchId
+    }
+    this.callback = callback
+  }
```
Comment (dajk) [inline src/matches.ts]: This whole thing seems a bit hacky and not very intuitive to me, can you explain?

**Comment by dajk** [comment]
> I think that this commit is good. Do you have somthing to add ?

I think this is already pretty good, good job. There is just that part that seems a bit fragile, or is it?

File: src/matches.ts (hunk 5 above)
Comment (honjes) [reply src/matches.ts]: Yes it could be smother but i dont know how else too do it without dropping support for the old system.

**Comment by honjes** [comment]
Do i have too do somthing too push this pullrequest ?

**Comment by dajk** [comment]
> Do i have too do somthing too push this pullrequest ?

Thank you @honjes for awesome work 🚀🚀🚀
I will bump up the minor version and release it these days.

**Pull request merged**